Extension/src/SSH/sshCommandToConfig.ts

```diff
@@ -0,0 +1,260 @@
+                );
+            }
+        }
+
+        /**
+         * For the cases of:
+         * 1. local_socket:host:hostport
+         * 2. local_socket:remote_socket
+         */
+
+        const delimiter: number = args.indexOf(':');
+        if (delimiter === -1) {
+            throw new CommandParseError(
+                `LocalFoward needs a listener and a destination separate by a colon. ${args} does not match.`
+            );
+        }
+
+        entries.LocalForward = `${args.substring(0, delimiter)} ${args.substring(delimiter + 1)}`;
+    },
+    l: (entries, user) => (entries.User = user),
+    M: entries => (entries.ControlMaster = 'yes'),
+    m: (entries, specs) => (entries.MACs = specs),
+    // -N (don't execute remote command) would interfere with execution
+    N: null,
+    // -n (redirect stdin) would interfere with execution
+    n: null,
+    o: (entries, option) => {
+        // Option keys never have equals signs in them, always fine to split the first
+        const delimiter: number = option.indexOf('=');
+        if (delimiter === -1) {
+            // `ssh` also fails if missing an argument (doesn't default to "yes", or anything)
+            throw new CommandParseError(`Argument missing for option ${option}`);
+        }
+
+        entries[option.slice(0, delimiter)] = option.slice(delimiter + 1);
+    },
+    p: (entries, port) => (entries.Port = port),
+    // -q (quiet mode) not useful for us
+    q: null,
+    R: (entries, address) => (entries.RemoteForward = address),
+    S: (entries, path) => (entries.ControlPath = path),
+    // -s (remote subsystem invokation), no setting in the config for this
+    s: null,
+    // -T (disable pseudo tty), no setting in the config for this
+    T: null,
+    // -t (enable pseudo tty), no setting in the config for this
+    t: null,
+    // -V (display the version) not relevant for us
+    V: null,
+    v: entries => (entries.LogLevel = 'verbose'),
+    W: (entries, address) => (entries.RemoteForward = address),
+    w: (entries, value) => (entries.TunnelDevice = value),
+    X: entries => (entries.ForwardX11 = 'yes'),
+    x: entries => (entries.ForwardX11 = 'no'),
+    Y: entries => (entries.ForwardX11Trusted = 'yes'),
+    // output logging
+    y: null
+};
+
+/**
+ * Directive passed to getopt(). From OpenSSH
+ * @see https://github.com/openssh/openssh-portable/blob/e3b6c966b79c3ea5d51b923c3bbdc41e13b96ea0/ssh.c#L662
+ */
+const getOptDirective: string = '1246ab:c:e:fgi:kl:m:no:p:qstvxAB:CD:E:F:GI:J:KL:MNO:PQ:R:S:TVw:W:XYy';
+
+/**
+ * Error thrown from sshCommandToConfig if the command could not be parsed.
+ */
+export class CommandParseError extends Error { }
+
+/**
+ * Attempts to convert an SSH command to an SSH config entry.
+ */
+export function sshCommandToConfig(command: string, name?: string): { [key: string]: string } {
+    const parts: string[] = parse(command) as string[];
+
+    // ignore 'ssh'	if the user entered that as their first word
```

Comment:
Extra spaces in comment?

Reply:
Probably randomly pressed some keys when I thought computer was freezed 🤣

Reply:
Actually it's a tab... So probably a typo.